### ISSUE TITLE
[BUG] Exception when using EmbedAuthor in Embed constructor

### ISSUE THREAD

**Issue by deadkex** [NAFTeam/NAFF#244]
**Describe the bug**
Exception when using EmbedAuthor in Embed constructor
`{'code': 50035, 'errors': {'data': {'embeds': {'0': {'author': {'name': {'_errors': [{'code': 'STRING_TYPE_CONVERT', 'message': 'Could not interpret "{\'name\': \'test\'}" as string.'}]}}}}}}, 'message': 'Invalid Form Body'}`

**To Reproduce**
```
embed = Embed(description="desc", author=EmbedAuthor("test"))
await ctx.send(embed=embed)
```

**Expected behaviour**
Be able to use EmbedAuthor in Embed constructor. Typing: Embed author: Optional[EmbedAuthor]

**Additional information**
Any other information about the problem.

**Comment by leestarb** [comment]
That's caused by incorrect `embed.to_dict` author property:
`{'description': 'desc.', 'author': {'name': {'name': 'me', 'url': 'https://example.com'}}}`

**Comment by benwoo1110** [comment]
Basically this issue is due to attrs converter. it converts the `EmbedAuthor(name="test")` into `EmbedAuthor(EmbedAuthor(name="test"))`

**Comment by LordOfPolls** [comment]
Fixed in 1ce602f5638ffd86fdc9a353881532977f03325e
